os/fs/driver/mtd/smart.c

```diff
@@ -186,13 +186,46 @@
 #if SMART_STATUS_VERSION == 1
 #define HEADER_IS_CLEAN(h) ((UINT8TOUINT16(h.logicalsector) == 0x0000 && h.seq == 0x00 && h.crc8 == 0x00 && h.status == 0x00))
 #elif SMART_STATUS_VERSION == 2
-#define HEADER_IS_CLEAN(h) ((UINT8TOUINT16(h.logicalsector) == 0x0000 && h.seq == 0x00 && h.crc16 == 0x0000 && h.status == 0x00))
+#define HEADER_IS_CLEAN(h) ((UINT8TOUINT16(h.logicalsector) == 0x0000 && h.seq == 0x00 && UINT8TOUINT16(h.crc16) == 0x0000 && h.status == 0x00))
 #elif SMART_STATUS_VERSION == 3
 #define HEADER_IS_CLEAN(h) ((UINT8TOUINT16(h.logicalsector) == 0x00000000 && h.seq == 0x00 && h.crc32 == 0x00000000 && h.status == 0x00))
 #endif
 
 #endif
 
+/*The following are CRC8 and CRC16 generated values which are constant 
+ *for a completely erased sector.These CRC values are indicative of the
+ *fact that the sector is indeed completely erased.
+ */
+
+#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF && SMART_STATUS_VERSION == 1
+#if CONFIG_MTD_SMART_SECTOR_SIZE == 256
+#define SMART_ERASEDSTATE_CRC  0
+#elif CONFIG_MTD_SMART_SECTOR_SIZE == 512
+#define SMART_ERASEDSTATE_CRC  255
+#elif CONFIG_MTD_SMART_SECTOR_SIZE == 1024
+#define SMART_ERASEDSTATE_CRC  99
```

Comment:
Could you share the method to calculate these values..?

Reply:
-The values for SMART_ERASEDSTATE_CRC are calculated using the CRC calculation function and standard table given in **_lib/libc/misc/lib_crc8.c_**.
-It makes use of a **_"0xA6" polynomial_** for CRC calculation.
-The values are put in as observed after the above mentioned steps were performed in function **_"smart_calc_sector_crc" (os/fs/driver/mtd/smart.c)_**.